### PR TITLE
fix(clippy): Fix clippy lints

### LINKED ISSUE #181: Fix clippy CI stage
## Description

Currently our Clippy CI stage doesn't work.

## Motivation

This is the way!

## Are you planning to do it yourself in a pull request?

Yes

### PR DESCRIPTION
**Pull request by Alex6323**
Closes #181

### CODE REVIEW

File: sdk/src/client/stronghold/storage.rs [hunk 1]
```diff
@@ -13,7 +13,7 @@ use crate::client::{storage::StorageProvider, Error, Result};
 
 #[async_trait]
 impl StorageProvider for StrongholdAdapter {
-    #[allow(clippy::significant_drop_tightening)]
+    // FIXME: #[allow(clippy::significant_drop_tightening)]
```
Comment (thibault-martinez) [inline sdk/src/client/stronghold/storage.rs]: Can you open an issue for this one please

File: .github/workflows/core.yml [hunk 2]
```diff
@@ -74,10 +74,10 @@ jobs:
       - name: Checkout the Source Code
         uses: actions/checkout@v3
 
-      - name: Set Up Beta Clippy
+      - name: Set Up Clippy
         uses: ./.github/actions/setup-rust
         with:
-          toolchain: beta
+          toolchain: stable
```
Comment (thibault-martinez) [inline .github/workflows/core.yml]: Not a big fan of this. I'd rather keep beta and temporarily allow if needed. Some clippy changes are pretty intense and may need some thinking about it. With stable we discover them too late.

Comment (Alex6323) [reply .github/workflows/core.yml]: yeah, I wanted to try different options (which is why this is just a draft atm)

File: sdk/src/wallet/account/operations/output_finder.rs [hunk 3]
```diff
@@ -154,7 +154,10 @@ impl AccountHandle {
 
             self.sync(sync_options).await?;
 
-            let output_count = self.read().await.unspent_outputs.len();
+            let output_count = {
+                let account = self.read().await;
+                account.unspent_outputs.len()
+            };
```
Comment (thibault-martinez) [inline sdk/src/wallet/account/operations/output_finder.rs]: What's that for ?

Comment (Alex6323) [reply sdk/src/wallet/account/operations/output_finder.rs]: lifetime shortening (that clippy understands)

Comment (Alex6323) [reply sdk/src/wallet/account/operations/output_finder.rs]: still a draft tho, I wouldn't review yet

Comment (thibault-martinez) [reply sdk/src/wallet/account/operations/output_finder.rs]: But the guard is only temporary here

Comment (Alex6323) [reply sdk/src/wallet/account/operations/output_finder.rs]: yeah, it's just that clippy can't see through it ... I changed it back and added an exception as this was perfectly fine Rust.

**Review by Thoralf-M — DISMISSED** [review]

File: sdk/src/wallet/bindings/java/lib/native/src/lib.rs [hunk 4]
```diff
@@ -76,6 +76,7 @@ macro_rules! env_assert {
 }
 
 // This keeps rust from "mangling" the name and making it unique for this crate.
+#[allow(clippy::missing_safety_doc)]
```
Comment (thibault-martinez) [inline sdk/src/wallet/bindings/java/lib/native/src/lib.rs:80]: Put TODOs before these, we should address them when we merge the Java bindings

**Review by thibault-martinez — APPROVED** [review]

**Review by Thoralf-M — APPROVED** [review]

**Review by thibault-martinez — CHANGES_REQUESTED** [review]
There is still one clippy error

**Review by thibault-martinez — DISMISSED** [review]

**Review by Thoralf-M — DISMISSED** [review]

**Review by Thoralf-M — APPROVED** [review]

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**